client/gutenberg/editor/controller.js

```diff
@@ -95,6 +97,21 @@ export const loadTranslations = ( context, next ) => {
 	} );
 };
 
+export const loadGutenbergBlockAvailability = async ( context, next ) => {
+	const { path } = context;
+	const siteFragment = getSiteFragment( path );
+
+	waitForData( {
+		blockAvailability: () => requestGutenbergBlockAvailability( siteFragment ),
+	} ).then( ( { blockAvailability } ) => {
+		if ( 'success' === blockAvailability.state && blockAvailability.data ) {
+			set( window, [ JETPACK_DATA_PATH, 'available_blocks' ], blockAvailability.data );
+		}
+
+		next();
+	} );
```

Comment:
if this fails then we're halting the page load I think - should we proceed with an error message instead? route to an error page?

Reply:
(This comment was referring to the original code that was here. Thanks to Dennis' #29166 on which I rebased this PR, it's now obsolete.)